src/temporal_terms.jl

```diff
@@ -0,0 +1,56 @@
+# Basic Lag:
+
+# syntax:
+"""
+    lag(term, nsteps)
+This `@formula` term is used to introduce lagged variables.
+For example `lag(x,1)` effectively adds a new column containing
+the value of the `x` column from the previous row.
+If there are no such row (e.g. because this is the first row), then the lagged column will
+contain `missing`.
+
+Note: this is only a basic row-wise lag operation.
+It is up to the user to ensure that that there data is sorted by the temporal variable,
+and that their observations are spaced with regular time-steps.
```

Comment:
```suggestion
and that observations are spaced with regular time-steps.
```
("the user" is singular)